src/stan/model/indexing/lvalue.hpp

```diff
@@ -693,11 +665,11 @@ inline void assign(
     Mat1&& x,
     const cons_index_list<Idx, cons_index_list<index_uni, nil_index_list>>&
         idxs,
-    const Mat2& y, const char* name = "ANON", int depth = 0) {
+    Mat2&& y, const char* name = "ANON", int depth = 0) {
   stan::math::check_range("matrix[..., uni] assign range", name, x.cols(),
                           idxs.tail_.head_.n_);
-  assign(x.col(idxs.tail_.head_.n_ - 1), index_list(idxs.head_), y, name,
-         depth + 1);
+  assign(x.col(idxs.tail_.head_.n_ - 1), index_list(idxs.head_),
+         std::forward<Mat2>(y), name, depth + 1);
```

Comment:
Matrix elements are always scalars, so there is no benefit from PF here.

Reply:
Sure, I posted yesterday on the Eigen issue tracker about this and assign to `.col()` with an rvalue still just uses copy semantics so I'm fine with changing this. (they could do some clever things but there's nothing there right now)

https://gitlab.com/libeigen/eigen/-/issues/2026

Reply:
Though if they do add those and we leave this then we don't have to change these things next time we upgrade Eigen

Reply:
Eh I'll get rid of them for now and we can keep track of what Eigen does

Reply:
Also I'd like to leave the `Mat2` input argument as `&&` just because everything else uses it and I think mix-match is more confusing then doing all one or all the other

Reply:
I disagree. Looking at function signature can tell you what the function can do with the argument. `const T&` is read-only, while `T&&` can be moved if it is a rvalue. What you are suggesting would make this overload look like it makes sense to forward arguments to it, even though it does not.

Reply:
Please also make the argument `const Mat2&`.

Reply:
Done!